compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1651,6 +1651,14 @@ fn check_fn_or_method<'tcx>(
     }
 }
 
+/// The `arbitrary_self_types_pointers` feature implies `arbitrary_self_types`.
+#[derive(Clone, Copy)]
+enum ArbitrarySelfTypesLevel {
+    None,                       // neither arbitrary_self_types nor arbitrary_self_types_pointers
```

Comment:
Nit: To me, this is often a sign you should remove this variant and use `Option<ArbirarySelfTypesLevel>` in the codebase. If you find yourself wanting the combinators that come with `Option`, consider refactoring this.